### PR TITLE
(DOCSP-14648): iOS Client Reset

### PR DESCRIPTION
**Pull request by cbush**
### Jira

https://jira.mongodb.org/browse/DOCSP-14648

### Staged Changes (Requires MongoDB Corp SSO)

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/client-reset/sdk/ios/advanced-guides/client-reset/

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/ios/advanced-guides/client-reset.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+- The client app made changes to that {+realm+} since the backup was made,
+  but did not sync those changes back to the server before the server
+  was restored.
+
+In other words, the client app must carry out a client reset on a given
+synced {+realm+} if the server is restored to a version older than the
+version on the client.
+
+A client reset erases all local data and downloads a new copy of the
```
Comment (nlarew) [inline source/sdk/ios/advanced-guides/client-reset.txt:31]: nit: could we define what a client reset is a bit earlier? IMO I'd rather have some idea of what a thing is before I know when it happens.

File: source/sdk/ios/advanced-guides/client-reset.txt [hunk 2]
```diff
@@ -0,0 +1,68 @@
+.. _ios-client-resets:
+.. _ios-client-reset:
+
+=======================
+Client Resets - iOS SDK
+=======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
```
Comment (nlarew) [inline source/sdk/ios/advanced-guides/client-reset.txt:10]: The contents directive is a bit weird when there's only one heading. Maybe we could add 1 or more headings?

File: source/sdk/ios/advanced-guides/client-reset.txt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+data stored in MongoDB Atlas. Performing a client reset loses all local
+changes made since the client last successfully synced.
+
+Example
+-------
+
+By default, the SDK performs client resets automatically on startup when
+instructed to do so by the server. When a client reset occurs, the SDK
+creates a backup of local data. By default, the SDK makes no effort to
+recover lost local changes from this backup.
```
Comment (nlarew) [inline source/sdk/ios/advanced-guides/client-reset.txt:41]: nit: this info seems more general than specific to this example

File: source/sdk/ios/advanced-guides/client-reset.txt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+If you override the client reset handler, you can access this backup
+copy through the path stored in the :objc-sdk:`SyncError
+<Enums/RLMSyncError.html#/c:@E@RLMSyncError@RLMSyncErrorClientResetError>`
+to manually transfer data from the backup copy to the newly created
+{+realm+}.
+
+.. tabs-realm-languages::
```
Comment (nlarew) [inline source/sdk/ios/advanced-guides/client-reset.txt:49]: nit: can we introduce the example? e.g. "The following example shows a custom client reset handler that..."

**Review by nlarew — APPROVED** [review]
A few comments but LGTM in general

File: source/sdk/ios/advanced-guides/client-reset.txt (hunk 3 above)
Comment (cbush) [reply source/sdk/ios/advanced-guides/client-reset.txt:41]: FYI you are commenting on content copied from other SDKs. Should we edit everywhere?

Comment (nlarew) [reply source/sdk/ios/advanced-guides/client-reset.txt:41]: Argh our duplication comes back to haunt us! One one hand yeah we probably should update them. Alternatively we can just merge this now and think about more holistic/cross-SDK updates later.

**Pull request merged**